### ISSUE TITLE
Consider using semantically meaningful icons for symbols

### ISSUE THREAD

**Issue by matklad** [lapce/lapce#2130]
Basically this vs code issue https://github.com/microsoft/vscode/issues/94263, but for lapce. Currently, icons are rather mysterious as to their meaning:

![image](https://user-images.githubusercontent.com/1711539/217882463-c9e2a699-770d-4a89-954e-410823e4ae28.png)

Using letters for icons makes them self-evident:

![image](https://user-images.githubusercontent.com/1711539/217882808-5f346ab6-985e-4ff8-af1b-3c7abe85b7fb.png)

**Comment by panekj** [comment]
You can replace icons with an icon theme, unfortunately both product icons and file icons are tied together but I'm hoping to change that soon

**Comment by matklad** [comment]
Yeah, I am eyeing this from a position of sane default. With VS Code, they inherited VS icons which are impossible to change due to inertia. Lance being new can experiment freely!

**Comment by panekj** [comment]
> Lance being new can experiment freely!

No, we can't, because iconography is difficult and we would have to create all icons ourselves or pay someone to do it, then we would have to extend icon set again when needed.
